frontend/src/routes/_layout/items.tsx

```diff
@@ -11,82 +11,73 @@ import {
   component: Items,
 })
 
-function Items() {
-  const showToast = useCustomToast()
-  const {
-    data: items,
-    isLoading,
-    isError,
-    error,
-  } = useQuery({
+function ItemsTable() {
+  const { data: items } = useSuspenseQuery({
     queryKey: ["items"],
     queryFn: () => ItemsService.readItems({}),
   })
+  return (
+    <TableContainer>
+      <Table size={{ base: "sm", md: "md" }}>
+        <Thead>
+          <Tr>
+            <Th>ID</Th>
+            <Th>Title</Th>
+            <Th>Description</Th>
+            <Th>Actions</Th>
+          </Tr>
+        </Thead>
+        <Tbody>
+          {items.data.map((item) => (
+            <Tr key={item.id}>
+              <Td>{item.id}</Td>
+              <Td>{item.title}</Td>
+              <Td color={!item.description ? "ui.dim" : "inherit"}>
+                {item.description || "N/A"}
+              </Td>
+              <Td>
+                <ActionsMenu type={"Item"} value={item} />
+              </Td>
+            </Tr>
+          ))}
+        </Tbody>
+      </Table>
+    </TableContainer>
+  )
+}
 
-  if (isError) {
-    const errDetail = (error as any).body?.detail
-    showToast("Something went wrong.", `${errDetail}`, "error")
-  }
-
+function Items() {
   return (
-    <>
-      {isLoading ? (
-        // TODO: Add skeleton
-        <Flex justify="center" align="center" height="100vh" width="full">
-          <Spinner size="xl" color="ui.main" />
-        </Flex>
-      ) : (
-        items && (
-          <Container maxW="full">
-            <Heading
-              size="lg"
-              textAlign={{ base: "center", md: "left" }}
-              pt={12}
-            >
-              Items Management
-            </Heading>
-            <Navbar type={"Item"} />
-            <TableContainer>
-              <Table size={{ base: "sm", md: "md" }}>
-                <Thead>
-                  <Tr>
-                    <Th>ID</Th>
-                    <Th>Title</Th>
-                    <Th>Description</Th>
-                    <Th>Actions</Th>
-                  </Tr>
-                </Thead>
-                <Tbody>
-                  {items.data.map((item) => (
-                    <Tr key={item.id}>
-                      <Td>{item.id}</Td>
-                      <Td>{item.title}</Td>
-                      <Td color={!item.description ? "ui.dim" : "inherit"}>
-                        {item.description || "N/A"}
-                      </Td>
-                      <Td>
-                        <ActionsMenu type={"Item"} value={item} />
-                      </Td>
-                    </Tr>
-                  ))}
-                </Tbody>
-              </Table>
-            </TableContainer>
-          </Container>
-        )
-      )}
-    </>
+    <Container maxW="full">
+      <Heading size="lg" textAlign={{ base: "center", md: "left" }} pt={12}>
+        Items Management
+      </Heading>
+
+      <ErrorBoundary fallback={<div>Something went wrong</div>}>
```

Comment:
@alejsdev what do you think of this?

I needed to add a new library, because you can't create error boundaries with functional components, and I thought it was better to import a new library rather than creating our own version of the component :D 

https://react.dev/reference/react/Suspense#providing-a-fallback-for-server-errors-and-client-only-content

https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary

Reply:
Nice, I like this approach :smile: